packages/contracts-bedrock/scripts/go-ffi/utils.go

```diff
@@ -127,3 +128,45 @@ func (n *proofList) Put(key []byte, value []byte) error {
 func (n *proofList) Delete(key []byte) error {
 	panic("not supported")
 }
+
+// encodeGasPayingToken encodes a gas paying token into a byte array.
+func encodeGasPayingToken(token common.Address, decimals uint8, name [32]byte, symbol [32]byte) ([]byte, error) {
+	tokenType, _ := abi.NewType("address", "", nil)
+	decimalsType, _ := abi.NewType("uint8", "", nil)
+	nameType, _ := abi.NewType("bytes32", "", nil)
+	symbolType, _ := abi.NewType("bytes32", "", nil)
+
```

Comment:
You don't need to define multiple `bytes32` types, you can reuse the same one multiple times. I also think that each of the types are created already and are in the global scope so you don't need to create types here, if `uint8Type` doesn't exist yet then you can add it to the place where they are all created globally

Reply:
See here: https://github.com/ethereum-optimism/optimism/blob/30f3027bac7231e3da3ba84e10ba98d1f8f6157e/packages/contracts-bedrock/scripts/go-ffi/differential-testing.go#L33